python/tests/graph/test_utils.py

```diff
@@ -0,0 +1,174 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+# Use this to create parameterized test cases
+from parameterized import parameterized
+
+import tensorflow as tf
+
+import sparkdl.graph.utils as tfx
+
+from ..tests import PythonUnitTestCase
+
+TestCase = namedtuple('TestCase', ['data', 'description'])
+
+
+def _gen_tensor_op_string_input_tests():
+    op_name = 'someOp'
+    for tnsr_idx in [0, 1, 2, 3, 5, 8, 15, 17]:
+        tnsr_name = '{}:{}'.format(op_name, tnsr_idx)
+        yield TestCase(data=(op_name, tfx.op_name(tnsr_name)),
+                       description='test tensor name to op name')
+        yield TestCase(data=(tnsr_name, tfx.tensor_name(tnsr_name)),
+                       description='test tensor name to tensor name')
+
+
+def _gen_invalid_tensor_or_op_input_with_wrong_types():
+    for wrong_val in [7, 1.2, tf.Graph()]:
+        yield TestCase(data=wrong_val, description='wrong type {}'.format(type(wrong_val)))
+
+
+def _gen_invalid_tensor_or_op_with_graph_pairing():
+    tnsr = tf.constant(1427.08, name='someConstOp')
+    other_graph = tf.Graph()
+    op_name = tnsr.op.name
+
+    # Test get_tensor and get_op with non-associated tensor/op and graph inputs
+    _comm_suffix = ' with wrong graph'
+    yield TestCase(data=lambda: tfx.get_op(tnsr, other_graph),
+                   description='test get_op from tensor' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_tensor(tnsr, other_graph),
+                   description='test get_tensor from tensor' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_op(tnsr.name, other_graph),
+                   description='test get_op from tensor name' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_tensor(tnsr.name, other_graph),
+                   description='test get_tensor from tensor name' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_op(tnsr.op, other_graph),
+                   description='test get_op from op' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_tensor(tnsr.op, other_graph),
+                   description='test get_tensor from op' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_op(op_name, other_graph),
+                   description='test get_op from op name' + _comm_suffix)
+    yield TestCase(data=lambda: tfx.get_tensor(op_name, other_graph),
+                   description='test get_tensor from op name' + _comm_suffix)
+
+
+def _gen_valid_tensor_op_input_combos():
+    op_name = 'someConstOp'
+    tnsr_name = '{}:0'.format(op_name)
+    tnsr = tf.constant(1427.08, name=op_name)
+    graph = tnsr.graph
+
+    # Test for op_name
+    yield TestCase(data=(op_name, tfx.op_name(tnsr)),
```

Comment:
Does it need to be generator-style instead of lists? Lists are easier to understand.